sbot/indexfeeds.go

```diff
@@ -120,44 +127,63 @@ func (manager indexFeedManager) Deregister(indexFeed refs.FeedRef) (bool, error)
 }
 
 // Process looks through all the registered indexes and publishes messages accordingly
-func (manager indexFeedManager) Process(m refs.Message) (refs.FeedRef, interface{}, error) {
+func (manager indexFeedManager) Process(m refs.Message) (refs.FeedRef, ssb.IndexedMessage, error) {
 	// creates index messages after actual messages have been published
 	var typed struct {
 		Type string
 	}
 	err := json.Unmarshal(m.ContentBytes(), &typed)
 	if err != nil {
-		return refs.FeedRef{}, nil, err
+		return refs.FeedRef{}, ssb.IndexedMessage{}, err
 	}
 
+	// do nothing (reasons for no type is usually encrypted messages)
 	if typed.Type == "" {
-		return refs.FeedRef{}, nil, nil
+		return refs.FeedRef{}, ssb.IndexedMessage{}, nil
 	}
 
 	// lookup correct index
 	idxKey := constructIndexKey(m.Author(), typed.Type)
 	pubkey, has := manager.indexes[idxKey] // use pubkey to later get publisher
 	if !has {
-		return refs.FeedRef{}, nil, nil
+		return refs.FeedRef{}, ssb.IndexedMessage{}, nil
 	}
 
-	content := indexMsg{
-		Type: "indexed",
-		Indexed: indexed{
-			m.Seq(),
-			m.Key(),
-		},
-	}
+	content := ssb.IndexedMessage{ Type: "indexed" }
+	content.Indexed.Sequence = m.Seq()
+	content.Indexed.Key=  m.Key()
 
+	manager.List() // TODO remove this cause it's just for testing
 	return pubkey, content, nil
 }
 
-// TODO: make sure this is the right format
-type indexed struct {
-	Sequence int64           `json:"sequence"`
-	Key      refs.MessageRef `json:"key"`
+func (manager indexFeedManager) List() ([]ssb.IndexListEntry, error) {
+	return manager.ListByType("")
 }
-type indexMsg struct {
-	Type    string  `json:"type"`
-	Indexed indexed `json:"indexed"`
+
+func (manager indexFeedManager) ListByType(msgType string) ([]ssb.IndexListEntry, error) {
+	indexlist := make([]ssb.IndexListEntry, 0, len(manager.indexes))
```

Comment:
if you want to be space/allocation efficient here, you should rather allocate the  whole thing, keep and index when filling and set the fields directly. Sadly the compiler isn't smart enough to figure out the way you create values below and append them can be done without copying them over.

```go
foo := make([]mystruct, 23}
fooIdx := 0

for ... := range {
  // filter etc
  foo[fooIdx].meta = m
  foo[fooIdx].stuff = s
  fooIdx++
}
```

Reply:
hmm that's a shame! thanks for letting me know c: i'll implement your suggested tweak (tho in this particular case it seems unlikely to matter tooo much from a performance pov?)

Reply:
you are right. this isn't perf ciritical but i've seen you doing this before.

as a matter of fact, in this case you could also leave out the `make([]stuff, 0, cap)` and just declare it as `var s []stuff` and use append like you did before and have it grow as needed with each loop.

Reply:
you are 100% correct i plead guilty to being attached to `append(list, item)` 😳

i appreciate you questioning that pattern, i think i'm walking away with a slightly better way of thinking about this type of situation now :3